actors/runtime/src/main/java/cloud/orbit/actors/runtime/KryoSerializer.java

```diff
@@ -483,9 +486,9 @@ public void serializeMessage(BasicRuntime basicRuntime, OutputStream outputStrea
                 writeHeaders(kryo, out, message.getHeaders());
                 writeNodeAddress(out, message.getFromNode());
                 writePayload(kryo, out, message);
-                return null;
-            }
-        });
+                return out.toBytes();
```

Comment:
no risk of data corruption here, new byte array is returned, not the pooled byte array.